core_impl/src/test/java/io/opencensus/implcore/tags/propagation/TagContextDeserializationTest.java

```diff
@@ -97,12 +99,17 @@ public void testDeserializeInvalidTagValue() throws Exception {
     os.write(SerializationUtils.VALUE_TYPE_STRING);
 
     // Encode a valid tag key and an invalid tag value:
-    encodeString("key", os);
+    encodeString("my key", os);
     encodeString("val\3", os);
     final byte[] bytes = os.toByteArray();
 
+    // TODO(sebright): Is there a better way to match a regular expression? 'Matches' extends
+    // Matcher<Object>, but 'expectMessage' takes a Matcher<String>.
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    Matcher<String> matcher = (Matcher) new Matches("Invalid tag value for key .*my key.*: val\3");
```

Comment:
Maybe something from this package can help.
http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/core/package-summary.html

Reply:
Why not jusr equals or prefix match? Do we need regex for this test?

Reply:
@bogdandrutu I found prefix, substring, and suffix matchers in the Hamcrest library, but not regex.

@adriancole I thought it would be better to not depend on the toString format used by Autovalue, but avoiding the exact match might not be worth the complexity.  I changed the test to use an exact match.